tests/optimize/test_momentum_qng.py

```diff
@@ -0,0 +1,342 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the QNG optimizer"""
+# pylint: disable=too-few-public-methods
+import pytest
+import scipy as sp
+
+import pennylane as qml
+from pennylane import numpy as np
+
+
+class TestExceptions:
+    """Test exceptions are raised for incorrect usage"""
+
+    def test_obj_func_not_a_qnode(self):
+        """Test that if the objective function is not a
+        QNode, an error is raised."""
+
+        dev = qml.device("default.qubit", wires=1)
+
+        @qml.qnode(dev)
+        def circuit(a):
+            qml.RX(a, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        def cost(a):
+            return circuit(a)
+
+        opt = qml.QNGOptimizer()
+        params = 0.5
+
+        with pytest.raises(
+            ValueError,
+            match="The objective function must be encoded as a single QNode.",
+        ):
+            opt.step(cost, params)
+
+
```

Comment:
It would be okay to skip this test because it tests code from `QNGOptimizer` that the subclass does not change :)
```suggestion
```